extension/scripts/background.js

```diff
@@ -19,4 +68,24 @@ chrome.runtime.onMessage.addListener(function(request, sender, sendResponse) {
 		enabled = request.enabled;
 		localStorage.setItem('enabled', String(request.enabled));
 	}
+
+	if (request.method === 'showWidget') {
+		chrome.tabs.query({
+			active: true,
+			lastFocusedWindow: true
+		}, function (tabs){
+			tabs.forEach(function (tab) {
+				chrome.tabs.sendMessage(tab.id, {method: 'showWidget'});
+			});
+		});
+	}
+
+	if (request.method === 'getData') {
+		co(() => getData(request.url))
+		.then(data => {
+			emitMessage('updateData', data, request.url);
+		}, e => {
+			emitMessage('updateError', {errorMessage: e.message}, request.url);
```

Comment:
I don't think we should surface the error message that the application generates to the end-user. Perhaps we could have a generic error message be used?